### PR TITLE
Implements AbstractFactory

### PR DESCRIPTION
**Pull request by MaxMEllon**
### About

I refactored to factory method pattern of GoF design pattern.
In particular, to made Motor Factory and Sensor Factory extends Abstract Factory.

### TODO

It has occurred a down cast, when create instance of sensors.
If possible, you should modify sensor interface to has abstract common methods.

### CODE REVIEW

File: src/jp/kagawa/etr/domain/factory/SensorFactory.java [hunk 1]
```diff
@@ -2,30 +2,33 @@
...
+		case COLOR_SENSOR:
+			return new ColorSensor(sensorPortValue.getColorSensorPort());
+		case GYRO_SENSOR:
+			return new GyroSensor(sensorPortValue.getGyroSensorPort());
+		case TOUCH_SENSOR:
+			return new TouchSensor(sensorPortValue.getTouchSensorPort());
+		case ULTRA_SONIC_SENSOR:
+			return new UltrasonicSensor(sensorPortValue.getSonarSensorPort());
+		default:
+			throw new IndexOutOfBoundsException("Out of index in MotorVariety");
```
Comment (MaxMEllon) [inline src/jp/kagawa/etr/domain/factory/SensorFactory.java]: Oops. This is don't to create Motor...

Comment (MaxMEllon) [inline src/jp/kagawa/etr/domain/factory/SensorFactory.java]: Fixed it

Comment (Yatsuku) [inline src/jp/kagawa/etr/domain/factory/SensorFactory.java]: You are using switch statements, but are you considering the process of downfalling each case?

Comment (MaxMEllon) [inline src/jp/kagawa/etr/domain/factory/SensorFactory.java]: used return statement, so no problem.

Comment (Yatsuku) [inline src/jp/kagawa/etr/domain/factory/SensorFactory.java]: ok, i'll try it.

**Comment by Yatsuku** [comment]
Oops.. . Althougn it operates,  the tail motor hops.

**Comment by Yatsuku** [comment]
It seems that there is no problem concerning program execution.
However, it seems necessary to investigate the problem of tail motor hopping.

**Pull request merged**